Dynamic page numbering for loop

slammajamma28 opened the pull request.

slammajamma28 committed:
Dynamic page numbering for loop

Vyrastas commented on ps3t_game_list.php:
```diff
@@ -96,7 +96,14 @@
+										  '/' . $gamealpha . '/');
+		$table_object = $init_html->find('div.bl_la_main div.divtext table', 1);
+		$page_object = $table_object->find('td', 0);
+		$page_num = explode(" ", $page_object)[4];
```
This works for PST but not for XBA... their pagination links are different: https://www.xboxachievements.com/browsegames/xbox-one/t/1

You'd have to do an if statement based on `$gamesite` to handle.

Vyrastas replied:
Also you should do a clear/unset for `$init_html` before you get into the for loop to free up memory (like at the bottom of the file for the `$html` variable).

slammajamma28 replied:
Is pagination actually different between the two?

https://www.playstationtrophies.org/browsegames/ps4/s/2
https://www.xboxachievements.com/browsegames/xbox-one/s/2

Based on the variables, it should work just fine for either... no?

slammajamma28 replied:
Ooooooooh oh oh oh I see what you mean now... XBA has the selectable numbers while PST has a "next page" option only. I'll see what I can do about that.

slammajamma28 replied:
Okay, so I also corrected the pagination. It now pulls the last page based on either PST or XBA

Vyrastas commented on ps3t_game_list.php:
```diff
@@ -96,7 +96,14 @@
         
         $nextpage = 1;
```
With a dynamic page number check we can remove anything that uses `$nextpage` and `$k` since we won't need to check if the next page exists.

slammajamma28 committed:
Requested Updates

Remove nextpage and k variables,
Unset/clear init_html

slammajamma28 committed:
Updated pagination

Will get correct "last page" since the pagination is different between PST and XBA

slammajamma28 commented:
Hopefully, the changes I made for dynamic page numbering work!

Vyrastas commented:
Sorry for the delay in looking at this... I'll verify that it works on my end tomorrow.

Vyrastas commented on ps3t_game_list.php:
```diff
@@ -161,21 +174,22 @@
         unset($html);
-        
+        $init_html->clear();
+        unset($init_html);
```
Let's put this clear/unset before line 112... once you get $page_num you don't need to retain the object for the rest of the script.

slammajamma28 replied:
Makes sense... smart!

Vyrastas commented on ps3t_game_list.php:
```diff
@@ -94,14 +94,27 @@
             echo '<B>' . str_replace('%20',' ',strtoupper($gametype[$i])) . '</B><BR>'; }
         
-        $nextpage = 1;
+        //$nextpage = 1;
```
Go ahead and remove all the commented out code, we don't need to retain it since we have the commit history.

slammajamma28 replied:
I removed ALL of the commented out code... didn't delete actual comments. :)

slammajamma28 committed:
Fixed formatting and requested changes

Moved init_html unset, removed comments
Also re-did indentation and removed whitespace at the end of lines.

slammajamma28 commented:
I also re-did for formatting, a lot of the indentation did not line up and it was driving me crazy. There were also a lot of instances of blank spaces at the end of the lines which I fixed.

Vyrastas commented:
Thanks for reformatting and cleaning up 👍 

So this works fine as far as I can tell, however — it slows down the game list page quite a bit.  Selecting a letter that previously took 6 seconds to return now takes 10 seconds.  I can push it to my site but it might be good to add a note that some letters may take a while to return.  Are you okay with that?

I'd also like to add a "processing" animation or something so people knowing it's working, but that can be done separately.

Vyrastas commented on ps3t_game_list.php:
```diff
@@ -91,94 +77,77 @@
+        if ($gamesite == 'ps3t') {
+            $table_object = $init_html->find('div.bl_la_main div.divtext table', 1);
+            $page_object = $table_object->find('td', 0);
+            $page_num = explode(" ", $page_object)[4]; }
```
```suggestion
            $page_num = substr(explode(" ", $page_object)[4], 0, -3); }

```

Vyrastas replied:
Tested some more... for PST you need to trim the last 3 characters for the pagenum, the way you had it returns:

"5"<br

Which works but let's make it cleaner.

slammajamma28 committed:
Update ps3t_game_list.php

Co-Authored-By: Jeff Bumgardner <jbumgardner@google.com>

slammajamma28 commented:
> So this works fine as far as I can tell, however — it slows down the game list page quite a bit. Selecting a letter that previously took 6 seconds to return now takes 10 seconds. I can push it to my site but it might be good to add a note that some letters may take a while to return. Are you okay with that?

I was just testing it out myself using my phone's stopwatch. I deployed both my code and your master code locally. 

For 'T', the master code took about 10 seconds and my code took about 11.5 seconds. 
For 'S', the master code took about 10.5 seconds and my code took 12.5 seconds. 

For some reason, the live site seems to pause before displaying anything while the local version appears to slowly display the text but the time for it to load is in line with the local master code times.

But yeah, I don't think the increase in wait time is too alarming.

> I'd also like to add a "processing" animation or something so people knowing it's working, but that can be done separately.

I agree, this would be a nice addition! In the meantime, a line of text could be added to ps3t_fr_include_alpha.php to state that some letters make take longer than others to display. I can also make a note on PST about the slight increase in wait time.

Vyrastas committed:
Merge branch 'master' into patch-1

Vyrastas commented:
Yeah, my site is a bit slower than running locally.  Not sure what the difference in progressive loading might be due to... PHP version?  Web server?  I don't know enough about how that works tbh.

I'll add a note about slow load times, then push the update to my website.

slammajamma28 commented:
Righteous! Believe it or not, this was my first legitimate pull request with actual review. It was a pleasure!

A few members at PST have requested an addition to the guide template which I’ll be working on... shouldn’t be as extravagant as dynamic page numbering... but we’ll see :)
